go-controller/pkg/node/gateway_udn.go

```diff
@@ -76,6 +78,11 @@ type UserDefinedNetworkGateway struct {
 	// iprules manager that creates and manages iprules for
 	// all UDNs. Must be accessed with a lock
 	ruleManager *iprulemanager.Controller
+
+	// isUDNNetworkAdvertised is a place holder to indicate whether the
+	// network is advertised or not
+	isUDNNetworkAdvertised     bool
+	isUDNNetworkAdvertisedLock sync.Mutex
```

Comment:
I don't think you need this, the information is already available in `util.NetInfo` above.